modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -125,8 +125,7 @@
             msg = sqlState + ": " + msg;
         }
 
-        if ((X.getCause(e) != null || !X.getSuppressedList(e).isEmpty())
-                && ctx.kernalContext().sqlListener().sendServerExceptionStackTraceToClient())
+        if (ctx.kernalContext().sqlListener().sendServerExceptionStackTraceToClient())
```

Comment:
Should we preserve `!X.getSuppressedList(e).isEmpty()` logic here?

Reply:
I'm not sure why. The option is called `sendServerExceptionStackTraceToClient`, so we just send the stack trace whatever it is, when enabled.

With those conditions in place I tried a few things like "disabled compute", "exception in compute task" and never got it to work.

Reply:
Ok, your solution is reasonable then